src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +67,272 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
+		// RC Pitch stick input for changing distance
+		const float distance_change_speed = FOLLOW_DISTANCE_USER_ADJUST_SPEED * _sticks.getPitch();
+		const float new_distance = _follow_distance + distance_change_speed * _deltatime;
+		follow_distance = constrain(new_distance, MINIMUM_DISTANCE_TO_TARGET_FOR_YAW_CONTROL, FOLLOW_DISTANCE_MAX);
+	}
+}
+
+void FlightTaskAutoFollowTarget::update_rc_adjusted_follow_angle(float &follow_angle, const float measured_orbit_angle, const float tracked_orbit_angle_setpoint) {
+	if (fabsf(_sticks.getRoll()) < USER_ADJUSTMENT_DEADZONE) {
+		return;
+	}
+	// Only apply Follow Angle adjustment if orbit angle setpoint and current orbit angle are within time window
+	// Wrap orbit angle difference, to get the shortest angle between them
+	if(fabsf(matrix::wrap_pi(measured_orbit_angle - tracked_orbit_angle_setpoint)) < FOLLOW_ANGLE_USER_ADJUST_SPEED * USER_ADJUSTMENT_ERROR_TIME_WINDOW) {
+		// RC Roll stick input for changing follow angle. When user commands RC stick input: +Roll (right), angle increases (clockwise)
+		// Constrain adjust speed [rad/s], so that drone can actually catch up. Otherwise, the follow angle
+		// command can be too ahead that it won't end up being responsive to RC stick inputs.
+		const float angle_adjust_speed_max = min(FOLLOW_ANGLE_USER_ADJUST_SPEED, _param_mpc_xy_vel_max.get() / _follow_distance);
+		const float angle_change_speed = angle_adjust_speed_max * _sticks.getRoll();
+		const float new_angle = _follow_angle_rad + angle_change_speed * _deltatime;
+		follow_angle = matrix::wrap_pi(new_angle);
+	}
+}
 
-		float offset_x = (float)cos(_follow_angle_filter.getState()) * _target_velocity_unit_vector(0) - (float)sin(
-					 _follow_angle_filter.getState()) * _target_velocity_unit_vector(1);
-		float offset_y = (float)sin(_follow_angle_filter.getState()) * _target_velocity_unit_vector(0) + (float)cos(
-					 _follow_angle_filter.getState()) * _target_velocity_unit_vector(1);
+void FlightTaskAutoFollowTarget::update_target_orientation(const Vector2f &target_velocity, float &current_target_orientation)
+{
+	const float target_velocity_norm = target_velocity.norm(); // Get 2D projected target speed [m/s]
 
-		// Lowpass on the offset vector to have smooth transitions when the target turns, or when the
-		// setting for the perspective is changed by the user. This introduces only a delay in the
-		// tracking / viewing angle without disadvantages
-		_offset_vector_filter.setParameters(_deltatime, DIRECTION_FILTER_ALPHA);
-		_offset_vector_filter.update(Vector2f{offset_x, offset_y});
+	// Depending on the target velocity, freeze or set new target orientatino value
+	if(target_velocity_norm >= TARGET_SPEED_DEADZONE_FOR_ORIENTATION_TRACKING) {
+		current_target_orientation = atan2f(target_velocity(1), target_velocity(0));
 	}
+}
+
+float FlightTaskAutoFollowTarget::update_orbit_angle(Vector2f &orbit_tangential_velocity, const float target_orientation, const float previous_orbit_angle_setpoint)
```

Comment:
I can create : 
1. generate_orbit_angle_trjaectory(), which returns the orbit angle, and then call
2. calculate_orbit_tangential_velocity, which would return calculation based off of the trajectory generated.

**Since TJ mentioned Silently changing internal variables is bad, I would keep a verbose function name like "update .. trajectory" for the first one because it implies that orbit angle trajectory has been generated (internally).**

But, then it's werid to return the new orbit angle, or should I name it : "genreate_orbit_anble_trajectory_and_get_orbit_angle"? 😃